creational/singleton.cr

```diff
@@ -4,16 +4,21 @@
 class Game
   getter name
 
-  @@instance = new
+  # @@instance = new
 
-  def initialize
+  # Game.new be prohibited outer.
+  private def initialize
     @name = "Mortal Kombat"
   end
 
+  # lazy instantiation
   def self.instance
-    return @@instance
+    @@instance ||= new
   end
 end
 
 puts Game.instance.name
+
+# private method 'new' called for Game:Class
+# game = Game.new
```

Comment:
`game = Game.new` is not valid code now and it may confuse people (at least it did me). Let's change these lines to:

```crystal
# private method 'new' called for Game:Class
puts Game.instance.name
# Mortal Kombat
```